### ISSUE TITLE
Contenttype field 'select' not working as child of 'block' type field

### ISSUE THREAD

**Issue by nout-kleef** [bolt/bolt#7062]
Named repeater fields containing a field with `type: select` always shows 

> No results found

when attempting to pick an option for said field.

 

Details
-------

 - Relevant Bolt Version: **3.4.0 beta 2**
 - Install type: **Composer install**
 - PHP version: **7.0.20**
 - Used web server: **Bolt built-in web server**
 - Database driver: **sqlite**

Reproduction
------------
### Bug summary

When a 
```yml
field:
    type: select
```
is nested in a
```yml
field:
    type: block
```
, the Bolt backend shows "No results found" when trying to pick an option for this select field.
I am not sure what could cause this issue to happen, but it most likely has to do with the fact that it only occurs when a select field is nested in a block field, and is therefore probably related to the new named repeaters.

### Specifics

url: http://0.0.0.0:8000/bolt/editcontent/cases/1
there are no hard error messages, the only way to notice the bug is when trying to pick an option.

### Steps to reproduce

Using the relatively new _named repeaters_ functionality as described [here](https://github.com/bolt/bolt/pull/6215),
with 
```yml
cases:
    name: Cases
    singular_name: Case
    fields:
        contentblocks:
            type: block
            fields:
                chapterblock:
                    fields:
                        chaptertype:
                            type: select
                            values: ['big', 'small']
```
as (relevant part of) contenttypes.yml, the bug occurs on multiple browsers (safari, chrome, firefox).

### Expected result

I expected an html input field that would allow me to choose from a dropdown list

### Actual result

, instead, the Bolt back end showed me the notice

> No results found

**Comment by GawainLynch** [comment]
In the linked PR, `contentblocks` is a ContentType named `contentblocks`, not an array of fields to be added to a ContentType.

```yaml
cases:
    name: Cases
    singular_name: Case
    fields:
        chapter_blocks:
            chapter_type:
                type: select
                values: ['big', 'small']
            chapter_binding:
                type: select
                values: ['hard', 'sort']
```

**Comment by nout-kleef** [comment]
Thank you for the quick reply.
I reviewed both the syntax in the original PR as well as the example you just commented, and the issue still exists in both examples.
Also, I wasn't sure where to put `type: block` in your example, so I tried it in a few locations, but all of them resulted in a Twig_Error_Runtime for the _imagelist_ variant, and a 

> No results found

for the _select_ variant.
Just in case you would like/need to take a look, here is my entire contenttype _cases_, which is fully functional except for the spots marked with ` # ISSUE`.
```yml
cases:
    name: Cases
    singular_name: Case
    fields:
        introtype:
            label: "Type intro scherm"
            type: select # working
            required: true
            values: ["Grote achtergrondafbeelding", "Kleinere afbeelding en logo"]
            postfix: "<p>Bepaal hoe het intro scherm eruit ziet</p>"
            group: content
        title:
            label: "Titel van de case"
            type: text
            required: true
            postfix: "<p>Wordt gebruikt voor de slug en op de pagina van de case</p>"
        logo:
            label: "Logo van de klant/het project"
            type: image
            attrib: [alt]
            required: false
            upload: logos
            postfix: "<p>Logo wordt alleen getoond indien 'Type intro scherm' == 'Kleinere afbeelding en logo'.</p>"
        introimage:
            label: "Afbeelding van het project"
            type: image
            attrib: [alt]
            required: true
            upload: intro-images
            postfix: "<p>Er zijn twee opties voor hoe deze afbeelding gebruikt wordt:<br /><strong>'Type intro scherm' == 'Grote achtergrondafbeelding'</strong>: gebruik deze afbeelding als achtergrond<br /><strong>'Type intro scherm' == 'Kleinere afbeelding en logo'</strong>: gebruik deze afbeelding als kleinere afbeelding bovenop een donkere achtergrond</p>"
        contentblocks:
            label: "Content Blokken"
            type: block
            fields:
                videoblock:
                    label: "Video block"
                    fields:
                        video:
                            label: "Video"
                            type: video
                            required: true
                            postfix: "<p>Voer hier de link naar een video in. NB: size heeft geen invloed op de grootte.</p>"
                figuresblock:
                    label: "Figures block"
                    fields:
                        figure1title:
                            label: "Naam figure 1"
                            type: "text"
                            required: true
                        figure1value:
                            label: "Waarde figure 1"
                            type: integer
                            required: true
                            default: 100
                        figure2title:
                            label: "Naam figure 1"
                            type: "text"
                            required: true
                        figure2value:
                            label: "Waarde figure 1"
                            type: integer
                            required: true
                            default: 100
                        figure3title:
                            label: "Naam figure 1"
                            type: "text"
                            required: true
                        figure3value:
                            label: "Waarde figure 1"
                            type: integer
                            required: true
                            default: 100
                        figure4title:
                            label: "Naam figure 1"
                            type: "text"
                            required: true
                        figure4value:
                            label: "Waarde figure 1"
                            type: integer
                            required: true
                            default: 100
                    postfix: "<p>Resultaten van het project. Deze waarden zullen in een oploopanimatie getoond worden.</p>"
                quoteblock:
                    label: "Quote/Title block"
                    fields:
                        quote:
                            label: "Quote"
                            type: text
                            required: true
                    postfix: "<p>Quote over/uitgangspunt van van het project </p>"
                chapterblock:
                    label: "Chapter block"
                    prefix: "<p>Tekstblokken voor op de case pagina. <strong>Worden als volgt gebruikt</strong>:<br />eerste chapter wordt gebruikt als 'vraagstuk van de klant',<br />tweede chapter wordt gebruikt als 'Wirelab oplossing',<br />alle volgende chapters worden later op de pagina achter elkaar geplaatst.</p>"
                    fields:
                        chaptertype:
                            label: "Type chapter"
                            type: select # ISSUE
                            # values: ['yes', 'no']
                            values: {'biggerPic': "Weinig tekst, grote afbeelding (3/4 schermbreedte)", 'smallerPic': "Meer tekst, kleinere afbeelding (1/2 schermbreedte)", 'textOnly': "Aleen tekst"}
                            postfix: "<p>Bepaal hoe dit chapter eruit ziet.</p>"
                        title:
                            label: "Titel"
                            type: text
                            required: true
                            postfix: "<p>Korte titel is aangeraden indien 'Type chapter' == 'Weinig tekst, grote afbeelding (3/4 schermbreedte)'</p>"
                        text:
                            label: "Tekst"
                            type: textarea
                            required: true
                            postfix: "<p>Om de afbeeldingen juist te weergeven, zullen de afbeeldingen niet hoger/langer worden wanneer er 'teveel' tekst aanwezig is.<br />Weinig tekst is daarom aangeraden indien 'Type chapter' == 'Weinig tekst, grote afbeelding (3/4 schermbreedte)'</p>"
                        expandable:
                            label: "Uitklapbaar"
                            type: select # ISSUE
                            values: {'yes': "Ja", 'no': "Nee"}
                            postfix: "<p>Dit stuk tekst uitklapbaar maken op mobile devices?<br />NB: vraagstuk en oplossing zijn standaard uitklapbaar</p>"
                        image:
                            label: "Afbeelding"
                            type: image
                            postfix: "<p>Kies een afbeelding die naast de tekst weergegeven kan worden indien 'Type chapter' != 'Alleen tekst'"
                        imagelocation:
                            label: "Locatie afbeelding"
                            type: select # ISSUE
                            values: {'left': "Links", 'right': "Rechts"}
                            postfix: "<p>Kies de locatie van de afbeelding die naast de tekst weergegeven wordt indien 'Type chapter' != 'Alleen tekst'"
                linkblock:
                    label: "'More about...' block"
                    postfix: "<p>Block om een button/link toe te voegen naar een landingspage of blogpost via sharpspring <strong>in progress...</strong></p>"
                    fields:
                        target:
                            label: "Link target"
                            type: select # ISSUE
                            required: false
                            values: deliverables/slug
                            postfix: "<p>Selecteer de url van de landingspage of blogpost waar deze link naar verwijst</p>"
                mobiledevicesblock:
                    label: "Mobile devices block"
                    fields:
                        color:
                            label: "Kleur van mobile devices"
                            type: select # ISSUE
                            values: ["Witte devices", "Zwarte devices"]
                            postfix: "<p>Specificeer de kleur van de iPhone en iPad</p>"
                        iphoneimage:
                            label: "iPhone afbeelding"
                            type: image
                            attrib: [alt]
                            upload: device-screenshots
                            required: true
                        ipadimage:
                            label: "iPad afbeelding"
                            type: image
                            attrib: [alt]
                            upload: device-screenshots
                            required: true
                    postfix: "<p>Kies afbeeldingen voor in de devices.</p>"
                macbookdeviceblock:
                    label: "Macbook device block"
                    fields:
                        image:
                            label: "Macbook afbeelding"
                            type: image
                            attrib: [alt]
                            upload: device-screenshots
                            required: true
                    postfix: "<p>Kies een afbeelding voor in het device.</p>"
                mobilescreenshotsblock:
                    label: "Mobile screenshots block"
                    fields:
                        shadows:
                            label: "Box shadows op afbeeldingen"
                            type: select # ISSUE
                            values: {'yes': "Wel shadows", 'no': "Geen shadows"}
                            postfix: "<p>Specificeer of de afbeeldingen een shadow moeten hebben</p>"
                        screenshot1:
                            label: "Eerste screenshot"
                            type: image
                            attrib: [alt]
                            upload: screenshots
                            required: true
                        screenshot2:
                            label: "Tweede screenshot"
                            type: image
                            attrib: [alt]
                            upload: screenshots
                            required: true
                        screenshot3:
                            label: "Derde screenshot"
                            type: image
                            attrib: [alt]
                            upload: screenshots
                            required: true
                tabletscreenshotsblock:
                    label: "Tablet screenshots block"
                    fields:
                        shadows:
                            label: "Box shadows op afbeeldingen"
                            type: select # ISSUE
                            values: {'yes': "Wel shadows", 'no': "Geen shadows"}
                            postfix: "<p>Specificeer of de afbeeldingen een shadow moeten hebben</p>"
                        count:
                            label: "Aantal tablet screenshots"
                            type: select # ISSUE
                            values: {'two': "Twee screenshots", 'four': "Vier screenshots"}
                        screenshot1:
                            label: "Eerste screenshot"
                            type: image
                            attrib: [alt]
                            upload: screenshots
                            required: true
                        screenshot2:
                            label: "Tweede screenshot"
                            type: image
                            attrib: [alt]
                            upload: screenshots
                            required: true
                        screenshot3:
                            label: "Derde screenshot"
                            type: image
                            attrib: [alt]
                            upload: screenshots
                        screenshot4:
                            label: "Vierde screenshot"
                            type: image
                            attrib: [alt]
                            upload: screenshots
                sliderblock:
                    label: "Image slider block"
                    fields:
                        images:
                            label: "Slider afbeeldingen"
                            type: imagelist # ISSUE
                            required: true
                            upload: slider-images
                            postfix: "<p>Afbeeldingen die zullen sliden om de 4 seconden en wanneer de gebruiker klikt.</p>"
                    prefix: "<p>NB: Afbeeldingen met gelijke hoogte geven het <strong>beste</strong> resultaat.</p>"
                finalimageblock:
                    label: "Final image block"
                    fields:
                        image:
                            label: "Afbeelding"
                            type: image
                            attrib: [alt]
                            required: true
                            upload: final-images
                    postfix: "<p>Kies een afbeelding die als laatste voor de footer op de pagina komt (full-width, 9:16)</p>"
            postfix: "<p>De knop hierboven kan gebruikt worden om de <strong>volgorde te bepalen</strong> van verschillende onderdelen van de website.<br />Geen inspiratie? De standaardvolgorde is:<br /><br /><strong>intro block</strong><br />chapter 1 (vraagstuk van klant)<br />motion video (vimeo)<br />chapter 2 (oplossing voor vraagstuk)<br />figures met optelanimatie<br />quote/titel<br />nog meer chapters<br />mobile devices<br />macbook<br />mobile screenshots<br />tablet screenshots<br />image slider<br />final image (full-width 9:16 afbeelding)<br /><strong>footer</strong><br /></p>"
        footer:
            label: "Footer cases"
            type: select # working
            values: cases/slug
            autocomplete: true
            sort: slug
            multiple: true
            postfix: "<p>Kies welke cases in de footer als suggesties getoond moeten worden.<br />Deze projecten worden <strong>alleen</strong> getoond indien er 3 projecten worden opgegeven.</p>"
        slug:
            label: "Slug"
            type: slug
            required: true
            uses: title
            group: meta
        previewimage:
            label: "Case afbeelding"
            type: image
            attrib: [alt]
            upload: preview-images
            postfix: "<p>Deze afbeelding wordt op de homepage als achtergrond getoond.<br />NB: dit is een <strong>fallback</strong> voor de 'Case video'.</p>"
        previewvideo:
            label: "Case video"
            type: video
            postfix: "<p>Een korte video die op de homepage wordt getoond op de achtergrond.<br />Deze video 'override' de 'Case afbeelding' indien beide gespecificeerd zijn.</p>"
        description:
            label: "SEO description"
            type: text
    record_template: case.twig
    taxonomy: [brands, partners, keywords]
    relations:
        deliverables:
            label: "Selecteer deliverables voor deze case"
            multiple: true
            postfix: "<p>Hier wordt naar verwezen vanuit 'deliverables' op de case pagina</p>"
        blogs:
            label: "Selecteer blogposts voor deze case"
            multiple: true
            postfix: "<p>Hier wordt naar verwezen vanuit 'deliverables' op de case pagina</p>"
    default_status: draft
```

as you can see the issue only exists for select and imagelist and not for any other field types, which is why I think it's not a syntactical problem in my `contenttypes.yml`

**Comment by GawainLynch** [comment]
Fixed in #7082
